tasmota/xdrv_23_zigbee_A_impl.ino

```diff
@@ -1331,6 +1331,36 @@ extern "C" {
       return 1;
     }
   }
+
+  void convert_seconds_to_dhms(uint32_t n, char *result){
+    int ret = 0;
+
+    uint32_t day = n / (24 * 3600);
+    n = n % (24 * 3600);
+    if(day > 0) {
+      ret = sprintf_P(result, PSTR("%02dd"), day);
+      result += ret;
+    }
+
+    uint32_t hour = n / 3600;
+    n %= 3600;
+    if(hour > 0) {
+      ret = sprintf_P(result, PSTR("%02dh"), hour);
+      result += ret;
+    }
+
+    uint32_t minutes = n / 60 ;
+    n %= 60;
+    if(minutes > 0) {
+      ret = sprintf_P(result, PSTR("%02dm"), minutes);
```

Comment:
and use
`snprintf_P(result, result_len, PSTR("%02dm"), minutes);`

Reply:
Refactored this entire section. See follow-up.